tools/codegen/codespec/api_to_provider_spec_mapper.go

```diff
@@ -2,69 +2,218 @@
+	if err != nil {
+		panic(err)
 	}
 
-	// TODO: remove after ToProviderSpecModel() implemented
-	return TestExampleCodeSpecification()
+	// map OAS resource model to CodeSpecModel
+	codeSpecResource := oasResourceToProviderSpecModel(oasResource, genConfig.Resources[resourceName], resourceName)
+
+	return &Model{Resources: []Resource{*codeSpecResource}}, nil
 }
 
-func TestExampleCodeSpecification() *Model {
-	testFieldDesc := "Test field description"
-	return &Model{
-		Resources: []Resource{{
-			Schema: &Schema{
-				Attributes: Attributes{
-					Attribute{
-						Name:                     "project_id",
-						ComputedOptionalRequired: Required,
-						String:                   &StringAttribute{},
-						Description:              conversion.StringPtr("Overridden project_id description"),
-					},
-					Attribute{
-						Name:                     "bucket_name",
-						ComputedOptionalRequired: Required,
-						String:                   &StringAttribute{},
-						Description:              &testFieldDesc,
-					},
-					Attribute{
-						Name:                     "iam_role_id",
-						ComputedOptionalRequired: Required,
-						String:                   &StringAttribute{},
-						Description:              &testFieldDesc,
-					},
-					Attribute{
-						Name:                     "state",
-						ComputedOptionalRequired: Computed,
-						String:                   &StringAttribute{},
-						Description:              &testFieldDesc,
-					},
-					Attribute{
-						Name:                     "prefix_path",
-						String:                   &StringAttribute{},
-						ComputedOptionalRequired: ComputedOptional,
-						Description:              &testFieldDesc,
-					},
-				},
-			},
-			Name: "test_resource",
-		},
-		},
+func oasResourceToProviderSpecModel(oasResource APISpecResource, resourceConfig config.Resource, name string) *Resource {
+	createOp := oasResource.CreateOp
+	readOp := oasResource.ReadOp
+
+	pathParamAttributes := pathParamsToAttributes(createOp, readOp)
+	createRequestAttributes := opRequestToAttributes(createOp)
+	createResponseAttributes := opResponseToAttributes(createOp)
+	readResponseAttributes := opResponseToAttributes(readOp)
+
+	attributes := mergeAttributes(pathParamAttributes, createRequestAttributes, createResponseAttributes, readResponseAttributes)
+
+	schema := &Schema{
+		Description:        oasResource.Description,
+		DeprecationMessage: oasResource.DeprecationMessage,
+		Attributes:         attributes,
 	}
+
+	resource := &Resource{
+		Name:   name,
+		Schema: schema,
+	}
+
+	applyConfigSchemaOptions(resourceConfig, resource)
+
+	return resource
+}
+
+func pathParamsToAttributes(createOp, readOp *high.Operation) Attributes {
+	pathParams := createOp.Parameters
+	pathParams = append(pathParams, readOp.Parameters...)
+
+	// TODO: fix path param description not mapping
+	pathAttributes := Attributes{}
+	for _, param := range pathParams {
+		if param.In != OASPathParam && param.In != OASQueryParam {
+			continue
+		}
```

Comment:
one detail, while path params are required query params can be optional. Would say for now we can focus on parsing path params.

Reply:
updated to use only pathParams for now